app/Services/PlanetService.php

```diff
@@ -53,30 +53,27 @@ class PlanetService
      *  If supplied the constructor will try to load the planet from the database.
      * @throws BindingResolutionException
      */
-    public function __construct(PlayerService $player = null, int $planet_id = 0)
+    public function __construct( PlayerService|null $player = null, int $planet_id = 0)
```

Comment:
Could you make the first param "?PlayerService" to indicate the null possible value, this is how its done in other places in this project.

Also please remove the space before the first param to make it `__construct(?PlayerService` instead of `__construct( ?PlayerService`. This to adhere to PSR-2 standards.